test/AccessController.ts

```diff
@@ -0,0 +1,95 @@
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { AccessControllerMock } from "../types/contracts/Mocks";
+import { signers } from "./ContractTestContext";
+
+describe("AccessController test suit", function () {
+    let owner: SignerWithAddress;
+    let addr1: SignerWithAddress;
+    let addr2: SignerWithAddress;
+    let addr3: SignerWithAddress;
+    let accessControllerMock: AccessControllerMock;
+    const initArg = 1234;
+
+    before(async () => {
+        [owner, addr1, addr2, addr3] = await signers;
+        /* AccessController is abstract so we need to use a contract that inherits it to test with: */
+        const factory = await ethers.getContractFactory("AccessControllerMock");
+        accessControllerMock = await factory.deploy(owner.address) as AccessControllerMock;
+    });
+
+    it("Should not allow non admin to init", async function () {
+        const initAsNonAdminPromise = accessControllerMock.connect(addr1).init(1234);
+        /* children with init need to add initializer modifier internally which will trigger this error */
+        await expect(initAsNonAdminPromise).to.be.revertedWith("Caller is not admin");
+    });
+
+    it("Should revert function with expectInitialized modifier if not initialized", async function () {
+        const initAsNonAdminPromise = accessControllerMock.mockExpectInitialized();
+        /* children with init need to add initializer modifier internally which will trigger this error */
+        await expect(initAsNonAdminPromise).to.be.revertedWith("Contract is not initialized");
+    });
+
+    it("Should allow admin to init", async function () {
+        await accessControllerMock.init(initArg);
+        const initialized = await accessControllerMock.getInitialized();
+        await expect(initialized).to.be.true;
+    });
+
+    it("Should execute code in child contract init function as part of AccessController.initializer", async function () {
+        /* someArg is set in mock child's init function: */
+        const someArg = await accessControllerMock.someArg();
+        expect(someArg).to.equal(initArg);
+    });
+
+    it("Should not allow admin to init twice", async function () {
```

Comment:
Whats the reasoning behind not allowing second init? things might change such as executive address etc etc